### PR TITLE
tests/benchmarks: add android_camera_preview benchmark test

### PR DESCRIPTION
**Pull request by ionela-voinescu**
Signed-off-by: Ionela Voinescu <ionela.voinescu@arm.com>

### CODE REVIEW

**Comment by ionela-voinescu** [comment]
@derkling in reply to your comment on https://github.com/ARM-software/lisa/pull/447:

The configuration (list of governors, list of tests, duration of benchmark, etc.) of tests in tests/benchmarks is usually there just as an example. Most of the times, in our tests, we change that to a configuration more appropriate to the purpose of our test.

File: tests/benchmarks/android_camera_preview.py [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+        try:
+            self.target.cpufreq.set_all_governors(self.governor);
+        except TargetError:
+            self._log.warning('Governor [%s] not available on target',
+                             self.governor)
+            raise
+
+        # Setup schedutil parameters
+        if self.governor == 'schedutil':
+            rate_limit_us = 2000
```
Comment (qperret) [inline tests/benchmarks/android_camera_preview.py]: Why not leaving this value as-is ? I'm not saying this is bad, and I can also see that we do the same for other benchmarks but what is the rationale ?

File: tests/benchmarks/android_camera_preview.py [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+                    tunables = self.target.cpufreq.get_governor_tunables(cpu_id)
+                    self.target.cpufreq.set_governor_tunables(
+                        cpu_id, 'ondemand',
+                        **{'sampling_rate' : tunables['sampling_rate_min']})
+            except TargetError as e:
+                self._log.warning('Failed to set ondemand parameters: {}'\
+                                 .format(e))
+                raise
+            self._log.info('Set ondemand.sampling_rate to minimum supported')
+
```
Comment (qperret) [inline tests/benchmarks/android_camera_preview.py]: What about other governors ? Should we enforce a specific value for `userspace` for ex ?

File: tests/benchmarks/android_camera_preview.py [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+# Run the benchmark in each of the supported governors
+
+duration_s = 20
+
+governors = [
+    'performance',
+    'ondemand',
+    'interactive',
+    'sched',
```
Comment (qperret) [inline tests/benchmarks/android_camera_preview.py]: Is this still needed ? Maybe for legacy ? schedfreq has been removed from recent android version (>=4.4)

File: tests/benchmarks/android_camera_preview.py [hunk 4]
```diff
@@ -0,0 +1,117 @@
+#!/usr/bin/env python
+
+import os
+from time import sleep
```
Comment (qperret) [inline tests/benchmarks/android_camera_preview.py]: Not used.

**Comment by valschneider** [comment]
Following a discussion with @ionela-voinescu, sending this to PR hell.

**Pull request closed without merging**